src/Aspire.Hosting.Kubernetes/KubernetesResourceContext.cs

```diff
@@ -0,0 +1,368 @@
+        {
+            if (volume.Source is null || volume.Target is null)
+            {
+                throw new InvalidOperationException("Volume source and target must be set");
+            }
+
+            if (volume.Type == ContainerMountType.BindMount)
+            {
+                throw new InvalidOperationException("Bind mounts are not supported by the Kubernetes publisher");
+            }
+
+            var newVolume = new VolumeMountV1
+            {
+                Name = volume.Source,
+                ReadOnly = volume.IsReadOnly,
+                MountPath = volume.Target,
+            };
+
+            Volumes.Add(newVolume);
+        }
+    }
+
+    private async Task ProcessArgumentsAsync(CancellationToken cancellationToken)
+    {
+        if (resource.TryGetAnnotationsOfType<CommandLineArgsCallbackAnnotation>(out var commandLineArgsCallbackAnnotations))
+        {
+            var context = new CommandLineArgsCallbackContext([], cancellationToken: cancellationToken);
+
+            foreach (var c in commandLineArgsCallbackAnnotations)
+            {
+                await c.Callback(context).ConfigureAwait(false);
+            }
+
+            foreach (var arg in context.Args)
+            {
+                var value = await ProcessValueAsync(arg).ConfigureAwait(false);
+
+                if (value is not string str)
+                {
+                    throw new NotSupportedException("Command line args must be strings");
+                }
+
+                Commands.Add(new(str));
+            }
+        }
+    }
+
+    private async Task ProcessEnvironmentAsync(DistributedApplicationExecutionContext executionContext, CancellationToken cancellationToken)
+    {
+        if (resource.TryGetAnnotationsOfType<EnvironmentCallbackAnnotation>(out var environmentCallbacks))
+        {
+            var context = new EnvironmentCallbackContext(executionContext, cancellationToken: cancellationToken);
+
+            foreach (var c in environmentCallbacks)
+            {
+                await c.Callback(context).ConfigureAwait(false);
+            }
+
+            var processedKeys = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+            foreach (var kv in context.EnvironmentVariables)
+            {
+                var value = await ProcessValueAsync(kv.Value).ConfigureAwait(false);
+                var key = kv.Key.ToManifestFriendlyResourceName();
+                var stringValue = value.ToString() ?? string.Empty;
+
+                if (processedKeys.Contains(key))
+                {
+                    continue;
+                }
+
+                // Move connection strings to secrets
+                if (key.IsConnectionString())
+                {
+                    var expression = key.ToHelmSecretExpression(resource.Name);
+                    Secrets[key] = new HelmExpressionValue(expression, stringValue);
+                    continue;
+                }
```

Comment:
I don't think we should do this. Only move values that themselves container secrets to secrets (secret parameters).

Reply:
I had that originally - the problem there though is ultimately this has to go in a configMap or a secret, and having it included in a configMap like all other env vars are didn't feel best, as its value contains references to other secrets (password for example).

Reply:
This is what we do in the azure container apps case. The value is ony secret if it has a secret (transitively). This removes all conventions around key names so that things like azure functions and other future conventions will always work. No need to look for a prefix.

Reply:
Makes sense ^^ - Will rework tomorrow to check the existing Secrets collection for an expression match on the contents of the value first.
6:30am now haha